### ISSUE TITLE
Can't use List<Object> in my document and can't query inside of it

### ISSUE THREAD

**Issue by mfalcier** [MorphiaOrg/morphia#1521]
**Describe the bug**
I've created an entity `Employee`, and a `Foo` class. 
The entity Employee contains an attribute with a list of Foos, `Foo` is not a document. 
```
@Entity("employees")
data class Employee (
    @Id
    var _id: String? = ObjectId().toString(),
    @Indexed
    var name: String? = null,
    @Indexed
    var foos: MutableList<Foo>? = mutableListOf()
)
```
```
data class Foo (
        @Indexed
        var test: String? = null
)
```
When I try to save it, I'm encountering an `org.bson.codecs.configuration.CodecConfigurationException` for the Foo class saying: _"Can't find a codec for class com.foo.Foo."_.

I kinda avoided this issue connecting to mongo **this** way:

```
val connectionString = ConnectionString("mongodb://127.0.0.1:27017")
val pojoCodecRegistry: CodecRegistry = CodecRegistries.fromProviders(PojoCodecProvider.builder().automatic(true).build())
val codecRegistry: CodecRegistry = CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(), pojoCodecRegistry)
val clientSettings = MongoClientSettings.builder()
        .applyConnectionString(connectionString)
        .codecRegistry(codecRegistry)
        .build()

val mongoClient = MongoClients.create(clientSettings)
val dataStore = Morphia.createDatastore(mongoClient, "morphia_test")
```

After this, I'm getting another error, since I want to query my document this way:
```
val result = dataStore.find(Employee::class.java)
                .filter(
                        eq("foos.test", "test1")
                )
```
The error is a `dev.morphia.query.ValidationException` saying: _"Could not resolve path 'foos.test' against 'com.foo.Employee'."_.

From here, I got stucked 2 days, and I couldn't find a workaround on it. As suggested [here](https://stackoverflow.com/questions/64574483/cant-query-nested-object-using-morphia-mongodb), I opened an issue.

**To Reproduce**
See the attached reproducer.
[reproducer.zip](https://github.com/MorphiaOrg/morphia/files/5457213/reproducer.zip)

**Expected behavior**
Not getting a CodecConfigurationException, since I was doing this on previous version of Morphia (1.3.2).
Not getting a ValidationException, since I was doing this on previous version of Morphia (1.3.2) and on daily basis on MongoDB Compass.

 - Server Version: Mongo Community 4.4, Atlas Cluster 4.2.10, Atlas Cluster 3.6.20 
 - Driver Version: Morphia 2.0.2 embedded one
 - Morphia Version: 2.0.2

**Comment by DefiantCatgirl** [comment]
For the `ValidationException`, I suspect that `@Embedded` may be required on `Foo`. 

However this is not a solution for the same `ValidationException` problem I'm having, since I cannot use `@Embedded` for embedded classes where I have to have an `@Id` field (as they are also used as separate entities)...

**Comment by mfalcier** [comment]
I've tried everything @DefiantCatgirl , without success :(
That is going to fix the validation, but it always result into an empty query result.

**Comment by evanchooly** [comment]
`@Embedded` is definitely required.  fwiw, you don't need to configure any codecs specifically like you're doing.  Morphia will take care of that for you.  I'll look in to the query bits today.

**Comment by evanchooly** [comment]
Running your reproducer, if I put `@Embedded` on `Foo` neither test fails and works as expected.

**Comment by mfalcier** [comment]
@evanchooly does It return a result? Because when I've tried It before, but the query didn't worked.
I'll try it later and let you know my results. Thanks.

**Comment by evanchooly** [comment]
Yes.  The query returns returns 1 item as expected.

**Comment by mfalcier** [comment]
Ok I've tried to put the `@Embedded` annotation on `Foo` and it worken on the first test. On the second it's not working, but I guess it's because I'm changin things on the codecs.
I've tried the annotation before, but probably only after cheating out with the codecs: since that annotation wasn't required in the past, I haven't thought it would solve the first `CodecConfigurationException`.
Thanks @evanchooly for your fast reply, and I guess the issue is solved.
